dvc/cache/base.py

```diff
@@ -700,12 +700,13 @@ def checkout(
         elif self.changed_cache(
             hash_info, path_info=path_info, filter_info=filter_info
         ):
-            if not quiet:
-                logger.warning(
-                    "Cache '%s' not found. File '%s' won't be created.",
-                    hash_info,
-                    path_info,
-                )
+            msg = (
+                "Cache entry with hash '%s' not found. File '%s' won't be "
+                "created."
+            )
+            logger.debug(
+                msg, hash_info.value, path_info,
```

Comment:
```suggestion
                msg, hash_info, path_info,
```

`HashInfo` has a `__str__` method now, and we don't want to drop the hash type from the message either, since it may not always be an `md5` depending on the remote/cache type.

Reply:
> In this case the real cause of the problem is not that the (local cache) checkout step failed, it's that the (remote) fetch step failed. It seems to me like what is really needed here for dvc fetch/pull would be a "is your remote up to date" hint (because someone else forgot to dvc push) rather than "is your (local) cache up to date" (because you forgot to dvc fetch/pull) hint.

I think you are right about the error message. I had not thought deeply about what might be a common cause of this error case, but you are right that it is likely that someone else forgetting to push is likely a common cause.

> HashInfo has a __str__ method now, and we don't want to drop the hash type from the message either, since it may not always be an md5 depending on the remote/cache type.

I had mistakenly assumed that only one hash type was used, but if that is not the case then we should include the hash type. I will need update the message format string though, as the change I made to the message doesn't make sense with the raw hash_info put in where just the hash was.

With that said, I had intentionally moved away from passing the raw hash_info object because I felt that the claim in #5248 that the message was too low level still applied even if we only printed the hash_info str representation in verbose mode (via only logging at DEBUG level). If this representation is OK to print out in verbose mode, then I agree it is fine to just go back to passing the hash_info object. What do you think @pmrowla?

Reply:
Keeping the hash type in the output should be preferred for the debug output because it is not always going to be an MD5 (it could also be an S3 or HTTP etag for example). The main complaint before was that printing the entire `repr()` for hash_info objects was too verbose, but it will be a more nicely formatted string in the form `md5: 12345...` now that https://github.com/iterative/dvc/pull/5319 is merged

Reply:
Ok, I am up to speed now! Thanks for pointing out the recent change.

Regardless, it sounds like more thought and additional changes are probably required to truly address the original issue, and I am not going to be able to get back to this for a bit. If it makes sense to avoid me sitting on this task without progress, I'm happy to close out this MR so that someone else can pick it up. Otherwise, I can pick it back up later (probably in a week or two).

Reply:
I'll close this for now then, but I'm not sure if anyone else will pick up this issue in the next couple of weeks. If it hasn't been addressed by then, feel free to re-open your PR whenever you have the time for it.